pkg/vm/engine/disttae/tools.go

```diff
@@ -1341,3 +1342,61 @@ func transferDecimal128val(a, b int64, oid types.T) (bool, any) {
 		return false, nil
 	}
 }
+
+func groupBlocksToObjects(blocks []*catalog.BlockInfo, dop int) ([][]*catalog.BlockInfo, []int) {
+	var infos [][]*catalog.BlockInfo
+	objMap := make(map[string]int, 0)
+	lenObjs := 0
+	for i := range blocks {
+		block := blocks[i]
+		objName := block.MetaLocation().Name().String()
+		if idx, ok := objMap[objName]; ok {
+			infos[idx] = append(infos[idx], block)
```

Comment:
Don't you need to sort here? Don't you need to ensure the order inside the object?